dotCMS/src/main/java/com/dotcms/vanityurl/cache/VanityUrlCacheImpl.java

```diff
@@ -0,0 +1,150 @@
+package com.dotcms.vanityurl.cache;
+
+import java.util.List;
+import java.util.Optional;
+import com.dotcms.vanityurl.model.CachedVanityUrl;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotCacheAdministrator;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.liferay.util.StringPool;
+import io.vavr.control.Try;
+
+/**
+ * This class implements {@link VanityUrlCache} the cache for Vanity URLs. Is used to map the Vanity
+ * URLs path to the Vanity URL content
+ *
+ * @author oswaldogallango
+ * @version 4.2.0
+ * @since June 24, 2017
+ */
+public class VanityUrlCacheImpl extends VanityUrlCache {
+
+    private DotCacheAdministrator cache;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Private field 'cache' could be made final; it is only initialized in the declaration or constructor.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)